services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/api/containers.py

```diff
@@ -61,7 +63,7 @@ def _format_result(container_inspect: dict[str, Any]) -> dict[str, Any]:
 
         return container_inspect
 
-    async with docker_client() as docker:
+    async with container_inspect_lock, docker_client() as docker:
         container_names = shared_store.container_names
```

Comment:
Q: why do you need a lock?

Reply:
There is this case, where the containers need to be restarted. The director-v2 directly forwards the restart call to the dy-sidecar. If the observation cycle is running, during the time of the container restart, it can receive an unexpected container status, this will cause the service to error and shut down.
The lock delays the result of the container-inspect.